lib/tapioca/cli.rb

```diff
@@ -130,6 +130,10 @@ def todo
       type: :string,
       desc: "The path to the Rails application",
       default: "."
+    option :halt_upon_load_error,
+      type: :boolean,
+      desc: "Halt upon a load error while loading the Rails application",
+      default: true
```

Comment:
Good question. I think originally we defaulted to rescuing the exceptions to support "edge case" apps that were unable to boot `config/application.rb` but still want RBI generation. For DSLs most of the compilers do depend on Rails but not necessarily all. An app could be in a strange state but they can still want `FrozenRecord` compiler to generate some RBIs. Without a rescue, gem loading would also fail.

> Do we really want to generate empty RBIs if the application didn't load?

Due to non-rails compilers we wouldn't be not generating anything in `tapioca dsl` so I can see the usecase.


I couldn't find any user reports but I feel like the use case above was an ask. [This](https://github.com/Shopify/tapioca/pull/5/commits/080675bb21a0a34d7c812a789ad5eb06a7cfc4d1) is where it was added as a safe_require so maybe @paracycle  has more context.

Reply:
I think we still need a flag, since users might not be able to "fix" the problem with their application being loaded, and if that is the case, then Tapioca is unable to generate anything at all. It could still be valuable for Tapioca to continue generating _something_ in those cases, rather than nothing.